### PR TITLE
Improve documentation, error reporting, modularity

### PR DESCRIPTION
**Pull request by sourque**

### CODE REVIEW

File: misc/id_linux.sh [hunk 1]
```diff
@@ -4,8 +4,5 @@ teamid=$(
 	zenity --entry= \
 		--text="Enter in your TeamID here"
 )
-if [[ ${#teamid} > 0 ]]; then
-	echo $teamid >/opt/aeacus/TeamID.txt
-else
-	notify-send -i /opt/aeacus/assets/logo.png "Aeacus SE" "No ID specified!"
-fi
+echo $teamid
```
Comment (safinsingh) [inline misc/id_linux.sh]: Is it not writing to the file anymore?

File: src/configs.go [hunk 2]
```diff
@@ -32,64 +41,72 @@ func writeConfig(sourceFile, destFile string) {
 		failPrint("Can't open scoring configuration file (" + sourceFile + "): " + err.Error())
 		os.Exit(1)
 	}
-	encryptedConfig := encryptConfig(configFile)
+	encryptedConfig, _ := encryptConfig(configFile)
```
Comment (safinsingh) [inline src/configs.go]: 🚨 Unhanded error alert 🚨

File: src/configs.go [hunk 3]
```diff
@@ -32,64 +41,72 @@ func writeConfig(sourceFile, destFile string) {
...
-	return decryptConfig(dataFile)
+	return decryptedConfig, nil
 }
 
+// printConfig offers a printed representation of the config, as parsed
+// by readData and parseConfig.
 func printConfig() {
 	passPrint("Configuration " + mc.DirPath + "scoring.conf" + " check passed!")
-	fmt.Printf("Title: %s (%s)\n", mc.Config.Title, mc.Config.Name)
+	fmt.Printf("Title: %s\n", mc.Config.Title)
```
Comment (safinsingh) [inline src/configs.go]: Why are you so obsessed with `Printf` :man_facepalming:

File: src/crypto.go [hunk 4]
```diff
@@ -45,11 +45,11 @@ func encryptConfig(plainText string) string {
 	writer.Close()
 
 	// XOR the encrypted file with our key.
-	return xor(key, encryptedFile.String())
+	return xor(key, encryptedFile.String()), nil
```
Comment (safinsingh) [inline src/crypto.go]: Why it just returning a nil error?

File: src/crypto.go [hunk 5]
```diff
@@ -59,15 +59,34 @@ func decryptConfig(cipherText string) string {
...
+	decryptedConfig := string(dataBuffer.Bytes())
+	if decryptedConfig == "" {
+		if debugEnabled {
+			failPrint("Scoring data is empty!")
+		}
+		return "", errors.New("Decrypted config is empty!")
+	}
 
-	return string(dataBuffer.Bytes())
+	return decryptedConfig, nil
```
Comment (safinsingh) [inline src/crypto.go]: Another nil error

File: src/gui_linux.go [hunk 6]
```diff
@@ -1,7 +1,12 @@
 package main
 
 func launchIDPrompt() {
-	// zenity prompt ree?
+	teamID, err := shellCommandOutput("bash /opt/aeacus/misc/id_linux.sh")
+	if err == nil {
+		writeFile("/opt/aeacus/TeamID.txt", teamID)
```
Comment (safinsingh) [inline src/gui_linux.go]: Ignore my previous comment about why it just echoes the team id

File: src/release_linux.go [hunk 7]
```diff
@@ -1,12 +1,14 @@
...
+// writeDesktopFiles creates TeamID.txt and its shortcut, as well as links
+// to the ScoringReport, ReadMe, and other needed files.
 func writeDesktopFiles() {
 	if verboseEnabled {
 		infoPrint("Creating or emptying TeamID.txt...")
 	}
-	shellCommand("touch /opt/aeacus/TeamID.txt")
 	shellCommand("echo 'YOUR-TEAMID-HERE' > /opt/aeacus/TeamID.txt")
 	shellCommand("chmod 666 /opt/aeacus/TeamID.txt")
+	shellCommand("chown " + mc.Config.User + ":" + mc.Config.User + " /opt/aeacus/TeamID.txt")
```
Comment (safinsingh) [inline src/release_linux.go]: Thank you :pray:

File: src/release_linux.go [hunk 8]
```diff
@@ -15,21 +17,28 @@ func writeDesktopFiles() {
...
-	gdm, err := pathExists("/etc/gdm/custom.conf")
-	if err != nil {
-		if lightdm {
-			writeFile("/usr/share/lightdm/lightdm.conf.d/50-ubuntu.conf", "echo autologin-user="+mc.Config.User)
-		} else if gdm {
-			writeFile("/etc/gdm/custom.conf", `echo AutomaticLogin=True
-AutomaticLogin=`+mc.Config.User)
-		} else {
-			failPrint("Unable to configure autologin. Please do so manually")
+	lightdm, _ := pathExists("/usr/share/lightdm")
```
Comment (safinsingh) [inline src/release_linux.go:24]: errs?

File: src/scoring.go [hunk 9]
```diff
@@ -64,79 +84,96 @@ func checkConfigData() {
 	readTeamID()
 }
 
+// scoreChecks runs through every check configured and runs them concurrently.
 func scoreChecks() {
 	mc.Image = imageData{}
-
 	assignPoints()
 
+	var wg sync.WaitGroup
```
Comment (safinsingh) [inline src/scoring.go:92]: Yayyyy

File: src/scoring.go [hunk 10]
```diff
@@ -64,79 +84,96 @@ func checkConfigData() {
...
 			}
-		} else {
-			if status {
-				if verboseEnabled {
-					failPrint(fmt.Sprintf("Penalty triggered: %s - %d pts", check.Message, check.Points))
-				}
-				mc.Image.Penalties = append(mc.Image.Penalties, scoreItem{check.Message, check.Points})
-				mc.Image.Score += check.Points
-				mc.Image.Detracts += check.Points
+			mc.Image.Points = append(mc.Image.Points, scoreItem{check.Message, check.Points})
```
Comment (safinsingh) [inline src/scoring.go:159]: extra spaces/tabs

File: src/utility.go [hunk 11]
```diff
@@ -2,16 +2,15 @@ package main
...
 	"io/ioutil"
-	"os"
 	"regexp"
-	"runtime"
-	"time"
 )
 
 const (
-	aeacusVersion = "1.3.0"
+	aeacusVersion = "1.4.0"
```
Comment (safinsingh) [inline src/utility.go:9]: :eyes:

File: src/gui_linux.go (hunk 6 above)
Comment (safinsingh) [reply src/gui_linux.go]: Also might wanna use `linuxDir` here

File: src/release_linux.go (hunk 7 above)
Comment (safinsingh) [reply src/release_linux.go]: Also `linuxDir` here too

**Review by safinsingh — CHANGES_REQUESTED** [review]
See comments

File: misc/id_linux.sh (hunk 1 above)
Comment (safinsingh) [reply misc/id_linux.sh]: Ignore this btw

File: src/release_linux.go (hunk 7 above)
Comment (sourque) [reply src/release_linux.go]: Good point, I'm just going to grep for `/opt/aeacus`

Comment (sourque) [reply src/release_linux.go]: There are however, still a lot of files that are not automatically generated (*desktop files) that reference the hardcoded path

Comment (safinsingh) [reply src/release_linux.go]: Hmm why can't we change that?

Comment (sourque) [reply src/release_linux.go]: We'd have to dynamically generate those files like we do with ScoringReport.html

Comment (safinsingh) [reply src/release_linux.go]: isn't it just `linuxDir + "/assets/ReadMe.html"`?

File: src/scoring.go (hunk 10 above)
Comment (sourque) [reply src/scoring.go:159]: I'm not sure what you mean, I `gofmt`'d it before pushing and it looks alright (?) to me.

File: src/release_linux.go (hunk 7 above)
Comment (sourque) [reply src/release_linux.go]: To reference them, sure, but in the files themselves they include the hardcoded `/opt/aeacus` dir

File: src/release_linux.go (hunk 8 above)
Comment (sourque) [reply src/release_linux.go:24]: `pathExists` never returns an error

Comment (safinsingh) [reply src/release_linux.go:24]: Oh then whats the other thing lol

File: src/crypto.go (hunk 4 above)
Comment (sourque) [reply src/crypto.go]: Well there were no errors returned in that function, but it turns out `zlib.Writer` does also return an error, so I'll check that.

File: src/crypto.go (hunk 5 above)
Comment (sourque) [reply src/crypto.go]: If it gets to that point, `err` is already `nil`, but I suppose it's good practice to return `err` (I think?)

Comment (safinsingh) [reply src/crypto.go]: Oh ok got it lol

File: src/configs.go (hunk 2 above)
Comment (sourque) [reply src/configs.go]: Has been ✋ handled ✋

**Review by safinsingh — DISMISSED** [review]
nvm :pensive:

**Review by safinsingh — APPROVED** [review]
😎😎😎

**Pull request merged**